### PR TITLE
chore: order components and config_items by path length to handle parent fkey

### PR DESCRIPTION
**Pull request by yashmehrotra**
Fixes: https://github.com/flanksource/duty/issues/404

### CODE REVIEW

File: upstream/upstream.go [hunk 1]
```diff
@@ -206,20 +207,25 @@ func (t *PushData) ApplyLabels(labels map[string]string) {
...
+		orderByClauses = append(orderByClauses, "LENGTH(COALESCE(path, ''))")
+	}
+	orderByClauses = append(orderByClauses, "id")
 	query := fmt.Sprintf(`
 		WITH p_keys AS (
  			SELECT id::TEXT, COALESCE(updated_at::text, '') as updated_at
 			FROM %s
 			WHERE id::TEXT > ? AND agent_id = ?
-			ORDER BY id
+			ORDER BY %s
```
Comment (adityathebe) [inline upstream/upstream.go:220]: I think we need this in descending order

Comment (yashmehrotra) [reply upstream/upstream.go:220]: No, both should be ascending

Comment (yashmehrotra) [reply upstream/upstream.go:219]: We need the ones with no path first, then 1 parent, then 2 parents and so on

```
                      SELECT id::TEXT, COALESCE(updated_at::text, '') as updated_at, path::text
                        FROM components  
                        ORDER by LENGTH(COALESCE(path, '')),id
                        LIMIT 500
;
                  id                  |          updated_at           |                                                      path                       
                               
--------------------------------------+-------------------------------+---------------------------------------------------------------------------------
-------------------------------
 01884897-1086-797c-137e-7e08cdd6f99e | 2024-02-16 04:42:05.036827+00 | 
 01884897-b6a6-96e4-5607-69962cf8844c | 2024-02-16 04:42:05.056775+00 | 
 018848a3-25b2-1114-2b23-1b038f6b2542 | 2023-07-10 09:19:55.379978+00 | 
 01884e0f-ab52-144b-408e-de44d73ca223 | 2024-02-15 22:26:05.199872+00 | 
 0189f817-9bb9-a8bc-4a1e-c19830134576 | 2024-01-11 08:07:39.481568+00 | 
 018b6582-004d-0a77-acb7-908e9108de02 | 2023-11-02 01:55:50.997271+00 | 
 018b89d7-b6e2-a3e8-7eb1-8ce550fe3fe4 | 2024-02-07 18:10:29.742821+00 | 
 018b89dd-a1db-4cd6-6246-b67b28638dd8 | 2023-12-29 06:07:16.741967+00 | 
 018b8a71-c8bb-e970-75da-af05d4ac1082 | 2024-02-16 04:42:05.138614+00 | 
 018caf15-ce1d-c2bc-992e-44e685c5f19c | 2024-01-11 08:32:39.258346+00 | 
 018cf300-87d0-9611-ffd1-c02c14e83b88 | 2024-01-11 08:36:54.286391+00 | 
 018cf78f-fdbd-1924-7682-1a610f944d2e | 2024-02-16 04:42:05.219757+00 | 
 018d55de-e756-8793-9b0f-c20fd00bc0fd | 2024-02-16 04:42:05.269106+00 | 
 01884897-108a-b5c3-5049-8d2ac31fb747 | 2024-02-02 07:19:01.15007+00  | 01884897-1086-797c-137e-7e08cdd6f99e
 01884897-108d-af4c-cc4c-13efe2ae8714 | 2024-02-02 07:08:53.703054+00 | 01884897-1086-797c-137e-7e08cdd6f99e
 01884897-1090-3d75-f809-f6995e6197a1 | 2024-02-02 11:41:20.333976+00 | 01884897-1086-797c-137e-7e08cdd6f99e
 01884897-1092-7e21-acf9-d924fe92f5de | 2024-02-02 11:37:03.963861+00 | 01884897-1086-797c-137e-7e08cdd6f99e
 01884897-b6ac-b8d1-97bb-f1332d4d18b0 | 2024-02-15 15:26:12.754923+00 | 01884897-b6a6-96e4-5607-69962cf8844c
 01884897-b6ae-3fda-c317-2fa9f5338e6f | 2024-02-16 04:36:13.034266+00 | 01884897-b6a6-96e4-5607-69962cf8844c
 01884897-b6b0-0818-9b85-cf1f9ac04e14 | 2024-02-02 07:27:06.917524+00 | 01884897-b6a6-96e4-5607-69962cf8844c
 01884897-b6b3-7980-093c-571373fd55b2 | 2024-02-16 04:31:10.897619+00 | 01884897-b6a6-96e4-5607-69962cf8844c
 01884897-b6b9-00e9-9242-e5605cc3a039 | 2024-02-16 04:41:11.04282+00  | 01884897-b6a6-96e4-5607-69962cf8844c
 ```

Comment (adityathebe) [reply upstream/upstream.go:219]: ah right! my bad.

**Review by adityathebe — APPROVED** [review]

**Pull request merged**